src/docs/release/breaking-changes/enterText-trailing-caret.md

```diff
@@ -0,0 +1,111 @@
+---
+title: `enterText` to move the caret to the end of the input text.
+description: `WidgetTester.enterText` and `TestTextInput.enterText` now move the caret to the end of the input text.
+---
+
+## Summary
+
+`WidgetTester.enterText` and `TestTextInput.enterText` now move the
+caret to the end of the input text.
+
+## Context
+
+The caret indicates the insertion point within the current text in an 
+active input field. Typically when a new character is entered, the 
+caret stays immediately after it. In Flutter the caret position is 
+represented by a collapsed selection. When the selection is invalid, 
+usually the user won't be able to modify or add text until they 
+change the selection to a valid value.
+
+`WidgetTester.enterText` and `TestTextInput.enterText` are 2 methods 
+used in tests to replace the content of the target text field. Prior 
+to this change, `WidgetTester.enterText` and `TestTextInput.enterText` 
+set the selection to an invalid range (-1, -1), indicating there's 
+no selection or caret. This contradicts the typical behavior of an
+input field.
+
+## Description of change
+
+In addition to replacing the text with the supplied text, 
+`WidgetTester.enterText` and `TestTextInput.enterText` now set the 
+selection to `TextSelection.collapsed(offset: text.length)`, instead
+of `TextSelection.collapsed(offset: -1)`.
+
+## Migration guide
+
+It should be very uncommon for tests to have to rely on the 
+previous behavior of `enterText`, since usually the selection 
+should not be invalid. **Consider changing the expected values of 
+your tests to adopt the `enterText` change.**
```

Comment:
you should include examples of error messages you might get here, so that googling for them reaches this page.

i would have expected a common failure to be golden file tests.

Reply:
Any keywords I should add to help people reach this page via googling? I feel it won't be particularly easy to find this by googling "golden test failure" or "caret moved to the end".